File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlRankClause.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression } from './ISqlExpression';

export class SqlRankClause implements ISqlExpression {
    SqlType: SqlType = SqlType.RankClause;
    Span: TextSpan = new TextSpan();
    
    Accept(visitor: any): void {
        // 簡單實作
    }

    ToSql(): string {
        return 'RANK()';
    }
} 

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlRankClause.cs
using T1.Standard.IO;

namespace T1.SqlSharp.Expressions;

public class SqlRankClause : ISqlExpression
{
    public SqlType SqlType { get; set; } = SqlType.RankClause;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_RankClause(this);
    }

    public ISqlExpression? PartitionBy { get; set; }
    public required ISqlExpression OrderBy { get; set; }

    public string ToSql()
    {
        var sql = new IndentStringBuilder();
        sql.Write("RANK() OVER (");
        if(PartitionBy != null)
        {
            sql.Write(PartitionBy.ToSql());
        }
        sql.Write(" " + OrderBy.ToSql());
        sql.Write(")");
        return sql.ToString();
    }
}